File: Datos/Interfaces/IUsuarioRepositorio.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Datos.ModeloDeDatos;
using Entidad.Modelos;
namespace Datos.Interfaces
{
   public interface IUsuarioRepositorio : IRepositorio<Usuario>, IDisposable
    {
        Usuario ObtenerUsuarioPorUserNameEmail(string documento);
        Usuario ObtenerPorDocumento(string documento);
        Usuario ObtenerPorID(int idUsuario);
        List<Usuario> GetAllUsuario();
        bool Obtener(string documento, string password, int idRolInvitado);
        List<MenuItemModel> ObtenerMenuUsuario(int idUsuario, MenuItemModel[] menuItems);
        
        List<MenuSidebar> ObtenerMenuUsuario(int idUsuario);

        void CambiarPassword(int idUsuario, string password);
        void ActualizarRolDeUsaurio(int idUsuario, int idRol, int idUsuarioLogueado);
        object logLogin(int idUsuario, string IP);
        Rol ObtenerRol(int idUsuario);
        void CreateUsuario(Usuario usuario);        
        void UpdateUsuario(Usuario usuario);




        //  List<Persona> BuscarPersona(string DNI, string apellido, string nombre, string sexo, string provincia, string localidad, string fechaNacimiento);
        // Persona obtenerPersonaPorId(int id);
        //  void AgregarAccion(Accion accion);
        //  Rol AgregarRol(Rol rol);                


        // List<Rol> ObtenerRoles();
        //void ActualizarUnidadDeUsaurio(int idPersona, int idUnidad, int idUsuarioLogueado);
        //List<Unidad> ObtenerUnidades();
        //Usuario PopularUsuario(int idPersona, int idRol, string password);
        //Persona CrearPersona(Persona personaParaGuardar, int idRol, string password);
        //Persona ActualizaPersona(Persona personaParaGuardar, int idRol, string password);
        //Persona ActualizaPersona(Persona personaParaGuardar);
        //Persona ObtenerPersonaPorIdPaciente(int idPaciete);
        //Persona ObtenerPersonaPorDocumento(int documento);
        //int ObtenerIdFuerzaPorDenominacion(string denominacion);
        //Localidad ObtenerInformacionLocalidadPorNombre(string nombreLocalidad);
        // Unidad ObtenerUnidadPorId(int id);


    }


}

File: Datos/Interfaces/IUsuarioRepositorio.vb
Imports HCI.Datos
Imports HCI.Entidades

Public Interface IUsuarioRepositorio
    Inherits IRepositorio(Of Usuario)
    Inherits IDisposable

    Function ObtenerPorDocumento(documento As Integer, idTipoDocumento As Integer) As Usuario

    Sub AgregarAccion(accion As Accione)


    Function ObtenerPorID(idUsuario As Integer) As Usuario


    Function AgregarRol(rol As Role) As Role

    Function ObtenerRol(idUsuario As Integer) As Role

    Overloads Function Obtener(documento As Integer, idTipoDocumento As Integer, password As String, idRolInvitado As Integer) As Boolean

    Function BuscarPersona(DNI As String, apellido As String, nombre As String, sexo As String, provincia As String, localidad As String, fechaNacimiento As String) As List(Of Persona)

    Function obtenerPersonaPorId(id As Integer) As Persona

    Function ObtenerRoles() As List(Of Role)

    Sub ActualizarRolDeUsaurio(idUsuario As Integer, idRol As Integer, idUsuarioLogueado As Integer)

    '@LL
    Sub ActualizarUnidadDeUsaurio(idPersona As Integer, idUnidad As Integer, idUsuarioLogueado As Integer)
    Function ObtenerMenuUsuario(idUsuario As Integer, menuItems As MenuItemModel())

    Function CambiarPassword(idUsuario As Integer, password As String)

    Function ObtenerUnidades() As List(Of Unidad)

    Function PopularUsuario(idPersona As Integer, idRol As Integer, password As String) As Usuario
    Function CrearPersona(personaParaGuardar As Persona, idRol As Integer, password As String) As Persona
    '@LL
    Function ActualizaPersona(personaParaGuardar As Persona, idRol As Integer, password As String) As Persona
    Function ActualizaPersona(personaParaGuardar As Persona) As Persona

    Function ObtenerPersonaPorIdPaciente(idPaciete As Integer) As Persona

    Function ObtenerPersonaPorDocumento(documento As Integer) As Persona

    Function ObtenerIdFuerzaPorDenominacion(denominacion As String) As Integer

    Function ObtenerInformacionLocalidadPorNombre(nombreLocalidad As String) As Localidad

    Function ObtenerUnidadPorId(id As Integer) As Unidad

    Function logLogin(idUsuario As Integer, IP As String) As Object


End Interface
